modules/service/src/main/scala/lucuma/odb/graphql/mapping/ProposalAttachmentMapping.scala

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.graphql
+
+package mapping
+
+import table.ProgramTable
+import table.ProposalAttachmentTable
+
+trait ProposalAttachmentMapping[F[_]] extends ProposalAttachmentTable[F] with ProgramTable[F] {
+
+  lazy val ProposalAttachmentMapping =
+    ObjectMapping(
+      tpe = ProposalAttachmentType,
+      fieldMappings = List(
+        // SqlField("program_id", ProposalAttachmentTable.ProgramId, key = true, hidden = true),
+        SqlField("program_id", ProposalAttachmentTable.ProgramId, hidden = true),
```

Comment:
The primary key in the table is `program_id` + `attachment_type`, but if I made both of those key fields here, in required my where clause in `ProposalAtttachmentMetadataService` to use both of these keys. I'm not sure if there are other consequences of not having `program_id` be a marked as a key here or which version is preferred.

Reply:
Yeah you need both key columns for sure, otherwise Grackle may not be able to tell them apart if you're querying across multiple programs. If this is a hassle you can create a view that constructs the composite key.